src/main/java/duke/command/DeletePatientCommand.java

```diff
@@ -51,33 +54,47 @@ public DeletePatientCommand(String deletedPatientInfo) throws DukeException {
      * @throws DukeException .
      */
     @Override
-    public void execute(PatientTaskList patientTask, TaskManager tasks, PatientManager patientManager,
+    public void execute(PatientTaskList patientTaskList, TaskManager taskManager, PatientManager patientManager,
                         Ui ui, PatientTaskStorage patientTaskStorage, TaskStorage taskStorage,
                         PatientStorage patientStorage) throws DukeException {
 
+        boolean toDelete;
+        ArrayList<PatientTask> patientTask = new ArrayList<>();
+        Patient patientToBeDeleted = null;
+
         if (id != 0) {
-            Patient patientToBeDeleted = patientManager.getPatient(id);
-            boolean toDelete = ui.confirmPatientToBeDeleted(patientToBeDeleted);
-            if (toDelete) {
-                patientManager.deletePatient(id);
-                ui.patientDeleted();
-                patientStorage.save(patientManager.getPatientList());
-            }
+            patientToBeDeleted = patientManager.getPatient(id);
+            ui.showPatientInfo(patientToBeDeleted);
         } else {
             ArrayList<Patient> patientsWithSameName = patientManager.getPatientByName(deletedPatientInfo);
             ui.patientsFoundByName(patientsWithSameName, deletedPatientInfo);
             if (patientsWithSameName.size() >= 1) {
                 int numberChosen = ui.choosePatientToDelete(patientsWithSameName.size());
                 if (numberChosen >= 1) {
-                    boolean toDelete = ui.confirmPatientToBeDeleted(patientsWithSameName.get(numberChosen - 1));
-                    if (toDelete) {
-                        patientManager.deletePatient(patientsWithSameName.get(numberChosen - 1).getID());
-                        ui.patientDeleted();
-                        patientStorage.save(patientManager.getPatientList());
-                    }
+                    patientToBeDeleted = patientsWithSameName.get(numberChosen - 1);
+                    ui.showPatientInfo(patientToBeDeleted);
                 }
             }
         }
+        try {
+            patientTask = patientTaskList.getPatientTask(patientToBeDeleted.getID());
+            ArrayList<Task> tempTask = new ArrayList<>();
+            for (PatientTask temppatientTask : patientTask) {
+                tempTask.add(taskManager.getTask(temppatientTask.getTaskID()));
+            }
+            ui.patientTaskFound(patientToBeDeleted, patientTask, tempTask);
+            toDelete = ui.confirmPatientToBeDeleted(patientToBeDeleted, true);
+        } catch (Exception e) {
+            toDelete = ui.confirmPatientToBeDeleted(patientToBeDeleted,false);
```

Comment:
You might considered to use a method provided by PatientTask class to get the list of PatientTasks assigned to a Patient.

Reply:
Actually I am using the method provided by PatientTask class to get the list of PatientTasks at line 80. From line 81 to 84, I am trying to get the list of tasks from taskManager that linked to each PatientTask I get from line 80.

Reply:
Yes you are right. Sorry for misunderstanding the code. There is no redundant code here.

Reply:
Yes you are right. Sorry for misunderstanding the code. There is no redundant code here.